doc/ug/sec_cm_dv_framework/index.md

```diff
@@ -0,0 +1,128 @@
+
+There are 2 requirements that are needed in this primitive.
+1. Create a standard primitive for each kind of countermeasure, such as a security counter module, which can detect if the counter value is modified unintentionally.
+These primitives could be reused across all blocks in the chip.
+Each primitive has an error indicator as an IO (or an internal signal if other blocks don’t need to connect with this error indicator).
+Once the design detects an attack, the error will be set.
+
+![diagram_of_sec_cm_prim](diagram_of_sec_cm_prim.svg)
+
+2. Once the error is set, the block should report a fatal alert or trigger unmasked interrupt to the system.
+There could be some other logics which are triggered when error occurs but not unified across the blocks.
+![diagram_of_block_that_contains_sec_cm_primitive](diagram_of_block_that_contains_sec_cm_primitive.svg)
+
+## Embedded Common Checks In The Design
+At the top of each IP, an assertion check should be added to ensure that the fatal alert will be triggered once error is set.
+In the countermeasure primitive, there is another check to ensure that each instance of the countermeasure primitive always has this assertion check.
+These 2 things can be accomplished by using the SystemVerilog macro.
+
+Firstly, define a macro in the countermeasure primitive and declare a logic “assert_connected” which will be assigned when the macro is invoked correctly.
+If the macro isn’t used for any instance of countermeasure primitive, the assertion will fail.
+
+```systemverilog
+`define ASSERT_PRIM_COUNT_ERROR_TRIGGER_ALERT(NAME_, PRIM_HIER_, ALERT_, MAX_CYCLES_ = 5) \
+  NAME_: assert property (@(posedge clk)) disable iff(!rst_n) \
+                ($rose(PRIM_HIER_.err_o) |-> ##[1:MAX_CYCLES_] $rose(ALERT_.alert_p)) \
+  `ifdef SIM_OR_FPV \
+  assign PRIM_HIER_.unused_assert_connected = 1'b1; \
+  `endif
+
+module prim_count();
+  ...
+  // This logic that will be assign to one, when user adds macro
+  // ASSERT_PRIM_COUNT_ERROR_TRIGGER_ALERT to check the error with alert, in case that prim_count
+  // is used in design without adding this assertion check.
+  `ifdef SIM_OR_FPV \
+  logic assert_connected;
+
+  initial #0 assert(assert_connected === 1'b1);
+  `endif
+endmodule
+```
+
+Secondly, invoke the assertion macro for each instance of the countermeasure primitive in the IP top.
+
+```systemverilog
+module a_ip_top();
+  ...
+  // Invoke the assertion macro for each instance of the countermeasure primitive
+  `ASSERT_PRIM_COUNT_ERROR_TRIGGER_ALERT(CtrlCntAlertCheck_A, u_ctrl.u_cnt, alert_tx_o[0])
+
+  `ASSERT_PRIM_COUNT_ERROR_TRIGGER_ALERT(FsmCntAlertCheck_A, u_fsm.u_cnt, alert_tx_o[0])
+
+endmodule
+
+```
+
+## Verification Framework For The Standardized Design Countermeasures
+
+This verification framework includes 3 types of verification works, which are all needed in order to acheive the completed verification for countermeasures.
```

Comment:
```suggestion
This verification framework involves three different steps, which are all needed in order to acheive the completed verification for countermeasures.
```
Or something like that? I find it is better to use a word for numbers in descriptions, and the sentence is repeating "verification *work*", which makes it weaker.